### DISCUSSION TITLE
How to strip query parameters from the resulting URL?

### DISCUSSION THREAD

**Discussion by otonote** [fastapi/fastapi#10242] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get('/Login')
async def Login(request:Request, response_model=HTMLResponse):
        # Do stuff with request.query_params here...

	return templates.TemplateResponse('Login.html', {'request': request})
```

### Description

I have this route which sometimes is called with query parameters. For example "/Login?error=Invalid password."

However, I don't want the end user to see those query parameters in their address bar. An end user that tries to bookmark this page will create a bookmark including the query parameters, which will create confusion.

How can I modify this code to strip out the query parameters from the resulting response? 

### Operating System

Windows

### FastAPI Version

0.103.1

### Pydantic Version

2.30

### Python Version

3.10.7

**Comment by chrisK824** [comment]
I guess you need to convert your HTTP method into a POST one which will just receive the body from another that serves the login form.

You cannot hide query parameters while using the GET method

**Comment by danielfcollier** [comment]
You can try to redirect the user to the same page without the query parameters:
https://fastapi.tiangolo.com/uk/advanced/custom-response/#redirectresponse

**Comment by YuriiMotov** [comment]
`request.url.remove_query_params(request.query_params.keys())._url` will return URL without query parameters
